docs/tasks.md

```diff
@@ -241,7 +241,24 @@ steps:
     #!/usr/bin/env bash
     /bin/my-binary
 ```
+#### Specifying a timeout
 
+A step can specify a `timeout` field. If the `Step` execution time exceeds the specified timeout, the `Step` and consequently the entire `TaskRun` is canceled.
+An accompanying error message is output under `status.conditions.message`.
+
+The format for a timeout is a duration string as specified in the [Go time package](https://golang.org/src/time/format.go?s=40541:40587#L1364) (e.g. 1s or 1ms).
+
+The example `Step` below is supposed to sleep for 60 seconds but will be canceled by the specified 5 second timeout.
+```yaml
+steps:
+  - name: willTimeout
+    image: ubuntu
+    script: | 
+      #!/usr/bin/env bash
+      echo "I am supposed to sleep for 60 seconds!"
+      sleep 60
+    timeout: 5s
```

Comment:
could this talk a bit about what happens when the timeout occurs? e.g. i assume the following steps don't execute, and is there anything in the logs to indicate that a timeout occurred?

Reply:
The first line [here](https://github.com/tektoncd/pipeline/blob/0c2ff5cd1a7fb47b2dc012fd99ddd19d9e3bf3d6/docs/tasks.md#specifying-a-timeout) mentions that `Step` and subsequently the entire `TaskRun` are canceled.

I'll elaborate and explicitly specify subsequent `Steps` won't execute too. 
I can elaborate a bit more on the timeout log as well.

Furthermore I'll amend the doc by mentioning that the process spawned by a `Step` is killed when a timeout occurs.